File: player.cpp
#include <string>
#include <iostream>
#include "player.h"
using namespace std;

Player::Player(){
	username = " ";
	password = " ";
	score = 0;
}

Player::Player(string u, string p, int s){
	username = u;
	password = p;
	score = s;
}

void Player::setUsername(string u){
	username = u;
}

void Player::setPassword(string p){
	password = p;
}

void Player::setScore(int s){
	score = s;
}

string Player::getUsername(){
	return username;
}

string Player::getPassword(){
	return password;
}

int Player::getScore(){
	return score;
}

void Player::printPlayer(){
	cout << "Username: " << username << endl;
	cout << "Password: " << password << endl;
	cout << "Score: " << score << endl;
}

File: player.js
class Player {
    constructor(u, p, s) {
        this.username = u || '';
        this.password = p || '';
        this.scores = [];
        this.highScore = s || 0;
    }

    setUsername(n) {
        this.username = n;
    }

    setPassword(p) {
        this.password = p;
    }

    addScore(s) {
        this.scores.push(s);
        if (s > this.highScore) {
            this.highScore = s;
        }
    }

    getUsername() {
        return this.username;
    }

    getPassword() {
        return this.password;
    }

    getHighScore() {
        return this.highScore;
    }

    printPlayer() {
        console.log(`Username: ${this.username}`);
        console.log(`Password: ${this.password}`);
        console.log(`High Score: ${this.highScore}`);
    }
}

module.exports = Player;
